src/main/java/org/springframework/data/elasticsearch/core/mapping/ElasticsearchPersistentEntity.java

```diff
@@ -171,9 +171,35 @@ default ElasticsearchPersistentProperty getRequiredSeqNoPrimaryTermProperty() {
 
 	/**
 	 * returns the default settings for an index.
-	 * 
+	 *
 	 * @return settings as {@link Document}
 	 * @since 4.1
 	 */
 	Document getDefaultSettings();
+
+	/**
+	 * @return the routing specification for the entity
+	 * @since 4.2
+	 */
+	@Nullable
+	String getRouting();
```

Comment:
Should return the evaluated expression (when using SpEL) or the routing string (when using literals).

Reply:
No. this is the value of the `@Routing` annotation that is defined on the entity class. Do you mean this should return a SpEL expression?

Reply:
I removed this method, the persistent entity now only has the `resolveRouting()` method, that now contains the complete logic.